Feature/ticpp3

petersteneteg opened the pull request:
Fixes #...

Changes proposed in this PR:
 * ...
 * ...
 * ...

This has been tested on: ...

> Thank you for contributing to the Inviwo repository! Please ensure the following: 
> - Code is error and warning free
> - Code follows the Inviwo coding guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Coding-Conventions
> - New Code / Processors follows the Inviwo documentation guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Documentation-guide
> - New Code / processors are tested in a unittest and / or regression test: 
>   https://github.com/inviwo/inviwo/wiki/Manual#testing
> 
> Please remember to specify the environment this code been compiled and tested on:
> Operating System / Compiler / IDE / QT / CMake / Python versions.

petersteneteg committed:
Core: avoid temp string when serializing

petersteneteg committed:
Ticpp: more PMR usage

petersteneteg committed:
Ticpp: more usage of pmr, avoid allocations.

# Conflicts:
#	include/inviwo/core/common/inviwomodule.h

petersteneteg committed:
Core: Serialiation optimizations

petersteneteg committed:
Core: WorkspaceManager use pmr when loading

petersteneteg committed:
Ticpp: Remove node locations. Only calculate location on errors

martinfalk commented on ext/ticpp/include/ticpp/element.h:
```diff
@@ -36,6 +37,11 @@ class TICPP_API TiXmlElement final : public TiXmlNode {
     */
     void SetAttribute(std::string_view name, std::string_view _value);
 
+    /** Adds an attribute of name to a given value. The attribute
```
```suggestion
    /** Adds an attribute name with the given value. The attribute
```

martinfalk committed:
Core,Ticpp: warn fixes

martinfalk committed:
Ticpp: alloc fix

petersteneteg committed:
Core: converter fix

petersteneteg committed:
Core: Clang Tidy fixes

# Conflicts:
#	ext/ticpp/include/ticpp/element.h
#	include/inviwo/core/util/inviwosetupinfo.h
#	src/core/io/serialization/serializer.cpp
#	src/core/network/workspacemanager.cpp

petersteneteg committed:
Core: merge fix

petersteneteg committed:
Core: typo fix

petersteneteg committed:
Core: type fix

martinfalk committed:
Core: compile fix
